.github/workflows/deploy-dev-v2.yml

```
@@ -0,0 +1,38 @@
+name: Deploy function to dev
+
+on:
+  push:
+    branches:
+      - feature-create-tree-v2 # On pushing to this branch, this workflow will be dispatched
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: "read"
+      id-token: "write"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: feature-create-tree-v2 # The branch to checkout for the deploy
```

I'll apply this change when approved, so that we move the github workflow to the main branch properly

```suggestion
      - main # On pushing to this branch, this workflow will be dispatched

jobs:
  build-and-deploy:
    runs-on: ubuntu-latest
    permissions:
      contents: "read"
      id-token: "write"
    steps:
      - name: Checkout
        uses: actions/checkout@v3
        with:
          ref: main # The branch to checkout for the deploy
```